mgmt: mcumgr: Add logging output

nordicjm opened the pull request:
Adds logging to mcumgr commands for debug and error reporting

Fixes #40850

~~Needs img_mgmt logging added and release notes updating~~

de-nordic reviewed (changes requested):
Nacking only because this deoends on https://github.com/zephyrproject-rtos/zephyr/issues/43696, which changes a lot of Kconfig option naming, and I want to avoid introducing Kconfig options just to rename them and deprecate.

nordicjm committed:
mgmt: mcumgr: Add logging output

Adds logging to mcumgr commands for debug and error reporting

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
doc: release-notes 3.3: Add information on MCUmgr logging

Adds details for new logging added to MCUmgr and Kconfig options to
control the level of reporting.

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>
